components/articles/cards/CardWrapper.js

```diff
@@ -1,28 +1,44 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import cn from 'classnames';
+import bem from 'bem-css-modules';
 
 import Link from 'components/common/Link';
 import BgContainer from 'components/common/ui/BgContainer';
 import { ThemeType } from 'utils/customPropTypes';
 import { colorLooksBlack, colorLooksWhite } from 'utils/ui';
 
+import styles from './cardWrapper.module.scss';
+
 export const DEFAULT_SIZES = ['xxl', 'xl', 'l', 'm'];
 export const TAG_SIZES = ['s', 'xs', 'ms', 's-wide', 'xs-wide'];
 export const SQUARE_SIZES = ['square-m', 'square-s'];
 export const SIZES = DEFAULT_SIZES.concat(SQUARE_SIZES, TAG_SIZES, 'auto');
 
-const CardWrapper = ({ className, size, children, image, color, theme, linkProps }) => (
-  <Link className={`card-size-${size}`} {...linkProps}>
+const b = bem(styles);
+
+const CardWrapper = ({
+  blockContextClass,
+  className,
+  size,
+  sizeClass,
+  children,
+  image,
+  color,
+  theme,
+  linkProps,
+  onBackground,
+}) => (
+  <Link className={size === 'auto' ? blockContextClass : sizeClass} {...linkProps}>
     <BgContainer
-      className={cn('wir-card-wrapper', className, {
-        'theme-light': theme === 'light',
-        // Despite of 'light' vs. 'dark' theming we also check if the card color is
-        // really close to black or white. We then add a border for blackish cards
-        // placed on top of black background; same for whitish.
-        'theme-white': colorLooksWhite(color),
-        'theme-black': colorLooksBlack(color),
-      })}
+      className={cn(
+        b({
+          'theme-light': theme === 'light',
+          'with-border':
+            (onBackground && colorLooksBlack(color)) || (!onBackground && colorLooksWhite(color)),
```

Comment:
`onBackground ? colorLooksBlack(color) : colorLooksWhite(color)`

Reply:
Right😂 I had a feeling I'm doing it wrong